handbook/engineering/developer-insights/code-insights/processes.md

```diff
@@ -16,28 +16,79 @@ Attendees are encouraged to add comments on talking points asynchronously before
-  - Work that is ready for development but not yet estimated or prioritized sits in the _Ready for Dev_ column.
-  - Anything that needs design input gets the `needs-design` label and goes in the _Needs design_ column.
-- When starting work, engineers pull cards from the _Planned_ column and move it to the _In Progress_ column.
-- There should never be more than a couple of cards in the _In Progress_ column at the same time. If we paused work on something (e.g. because priorities changed), it should be moved back to _Planned_ or _Icebox_ as appropriate.
-- Things we triaged from the _Inbox_, but don't plan to work on, go into the _Icebox_.
+Our current process is as follows:
 
-Note: we intentionally plan so we can avoid merging significant work less than two days before a release (if a release is on the 20th, our last day to merge is the 18th). Exceptions require explicit approval of both the PM and EM.
+- We work in **2-week iterations**. Every iteration as a GitHub milestone, which is created at least one iteration in advance.
 
-## Tracking Issues
+- Incoming tickets (from team members or from other teams) are added to the [GitHub project](https://github.com/orgs/sourcegraph/projects/200/views/1) and will be visible in the "All iterations" view under "No milestone". This represents our **backlog** of _unscheduled_ work.
 
-The team makes use of [tracking issues](../../tracking_issues.md) for tracking progress on the implementation of longer projects that need to be broken up into multiple issues.
-The teammates should ensure that a tracking issue is created when starting work on features that are expected to take longer than a few days (or require multiple PRs) to deliver.
+- **While an iteration is ongoing, we plan the next iteration**. This is mostly a mostly asynchronous process.
+
+  - Engineers, designer, PM and EM **_propose_ issues to be worked on** the week _before_ the mid-iteration [Monday sync](#weekly-sync) by adding them to the next iteration milestone and setting "Status" to "Proposed". This is done considering our [higher-level goals and roadmap](./goals.md).
+
+    - The issue should also have its **_Estimate_ column filled out**, so that it can be evaluated whether it fits into the iteration. If the proposer lacks the information to estimate the issue, they reply on the issue in GitHub or raise it in our Slack channel to get the missing information or get an estimate from the appropiate person. Teammates may also discuss this in ad-hoc synchronous meetings if beneficial. An assignee may also already volunteer or be proposed, but this may still be changed at the [Monday sync](#weekly-sync) to distribute workload.
+      - If **technical exploration** is needed to get more information, a _spike_ (a time-boxed investigation task meant to facilitate more granular planning) can be proposed for the next iteration instead to get that information.
```

Comment:
Minor – what if we both need a spike and otherwise need to complete both spike + feature in the iteration? (Let's say the estimation uncertainty is "anywhere from 2-5 days"; can we just slate both spike and feature for that iteration?)

Reply:
I think that's possible. Do you think there's value in documenting that? I'm a little concerned that this process becomes very long to read through/overwhelming

Reply:
Agree with @felixfbecker here. I don't think we need to explicitly state that we may have exceptions. During the sync meetings if something comes up we should be able to just say it then and move on.

Reply:
Sounds good!